scripts/pipeline/orthology_benchmark.py

```diff
@@ -17,18 +17,134 @@
 
 Typical usage example::
 
-    $ python orthology_benchmark.py --mlss_conf /path/to/mlss_conf.xml \
-        --species_set default
+    $ python orthology_benchmark.py --mlss_conf /path/to/mlss_conf.xml --species_set name \
+    --host mysql-ens-compara-prod-X --port XXXX --user ensro --out_dir /path/to/out/dir
 
 """
 
 import argparse
+import os
+import subprocess
+from typing import Dict, List
+
+import MySQLdb
+import numpy
 
 from ensembl.compara.config import get_species_set_by_name
 
 
-def dump_genomes():
-    """Docstring"""
+def dump_genomes(species_list: List[str], species_set_name: str, host: str, port: int,
+                 user: str, out_dir: str) -> None:
+    """Dumps canonical peptides of protein-coding genes for a specified list of species
+    from the latest available core databases to FASTA files using `dump_gene_set_from_core.pl`.
+
+    Args:
+        species_list: A list of species (genome) names to dump.
+        species_set_name: Species set (collection) name.
+        host: Database host.
+        port: Host port.
+        user: Server username.
+        out_dir: Directory to place `species_set_name/core_name.fasta` dumps.
+
+    Raises:
+        FileExistsError: If directory `out_dir/species_set_name` already exists.
+        MySQLdb.OperationalError: If connection to `host:port` cannot be established.
+        RuntimeError: If no cores are found for any species in `species_list` (`species_set_name`)
+                on `host:port`.
+
+    """
+    # Check access to the host & port
+    MySQLdb.connect(host=host, port=port, user=user)
```

Comment:
So I have just tested and it seems you can do:
```
from sqlalchemy import create_engine
eng = create_engine('mysql://ensro@mysql-ens-compara-prod-10:4648/')
eng.execute("SHOW DATABASES LIKE '%%_core_%%'").fetchall()
```
And you will get all core databases.

My concern about doing this is not about doing 1 search and then filtering (in most cases this will be more efficient), but with the fact that you are not specifying the release. If you are running your code, say for `vertannot`, you expect e105 cores and genebuild adds one e106 of one of the species you are searching for just before you run your code, you will end up with a different set of species. Not sure how relevant this is for your goal, but just mentioning a possible scenario.

Reply:
I'm aware that not all cores might be from the same release. That's favourable in situations when some species get retired/removed or if a core for newly available species is ready before others. There can be an option to specify a release, if needed.

Reply:
I like your idea Thomas 🙂  I had thought about it but very briefly. Now when I see all the pros listed at one place, I think I have to do it 🙂  Thank you very much 👍